addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,326 @@
-        for path in drivers:
-            result += "<li>" + path + ":" + str(drivers[path].value) + "</li>"
-        result += "</ul>"
-        result +=" </body></html>"
-        return result
-
-    @http.route('/hw_drivers/driverdetails/<string:identifier>', type='http', auth='none', cors='*')
-    def statusdetail(self, identifier):
-        for device in drivers:
-            if device.find(identifier) != -1:
-                return str(drivers[device].value)
-        return ''
-
-    @http.route('/hw_drivers/driveraction/<string:identifier>', type='json', auth='none', cors='*', csrf=False)
-    def driveraction(self, identifier):
-        data = httprequest.jsonrequest
-        result = 'device not found'
-        if data.get('action') == 'print':
-            with open('/tmp/toprinter', 'w') as file:
-                file.write(data['data'])
-            subprocess.call("cat /tmp/toprinter | base64 -d | lp -d " + identifier, shell=True)
-            result = 'ok'
-        if data.get('action') == 'camera':
-            cameras = subprocess.check_output("v4l2-ctl --list-devices", shell=True).decode('utf-8').split('\n\n')
-            adrress = '/dev/video0'
-            for camera in cameras:
-                if camera:
-                    camera = camera.split('\n\t')
-                    serial = re.sub('[^a-zA-Z0-9 ]+', '', camera[0].split(': ')[0]).replace(' ','_')
-                    if serial == data.get('identifier'):
-                        adrress = camera[1]
-            picture = subprocess.check_output("v4l2-ctl --list-formats-ext|grep 'Size'|awk '{print $3}'|sort -rn|awk NR==1", shell=True).decode('utf-8')
-            subprocess.call("fswebcam -d " + adrress + " /tmp/testimage -r " + picture, shell=True)
-            image_bytes = subprocess.check_output('cat /tmp/testimage | base64',shell=True)
-            result = {'image': image_bytes}
-        return result
-
-    @http.route('/hw_drivers/send_iot_box', type='http', auth='none', cors='*')
-    def send_iot_box(self):
-        send_iot_box_device(False)
-        return 'ok'
 
 #----------------------------------------------------------
-# Driver common interface
+# Helper
 #----------------------------------------------------------
-class Driver(Thread):
-    ping_value = ""
 
-    def supported(self):
-        pass
+def get_mac_address():
+    try:
+        return netifaces.ifaddresses('eth0')[netifaces.AF_LINK][0]['addr']
+    except:
+        return netifaces.ifaddresses('wlan0')[netifaces.AF_LINK][0]['addr']
 
-    def value(self):
-        pass
+def get_ip():
+    try:
+        return netifaces.ifaddresses('eth0')[netifaces.AF_INET][0]['addr']
+    except:
+        return netifaces.ifaddresses('wlan0')[netifaces.AF_INET][0]['addr']
 
-    def get_name(self):
-        pass
+def read_file_first_line(filename):
+    content = ""
+    try:
+        path = os.getenv("HOME") + '/' + filename
```

Comment:
you probably better os path join then is file then with open(file, 'r')

Reply:
from os.path import join as opj

In [5]: opj(os.getenv("HOME"), 'coucou')
Out[5]: '/Users/odoo/coucou'
In [8]: os.path.isfile(os.path.join(os.getenv('HOME'), 'coucou'))
Out[8]: False

In [9]: os.path.isfile(os.path.join(os.getenv('HOME'), 'jobs.txt'))
Out[9]: True